src/extra/libs/rlottie/lv_rlottie.c

```diff
@@ -155,14 +178,84 @@ static void lv_rlottie_destructor(const lv_obj_class_t * class_p, lv_obj_t * obj
 
 }
 
+#if LV_COLOR_DEPTH == 16
+static void convert_to_rgba5658(uint8_t * pix, const size_t width, const size_t height)
+{
+    /* rlottie draws in ARGB32 format, but LVGL only deal with RGB565 format with (optional 8 bit alpha channel)
+       so convert in place here the received buffer to LVGL format. */
+    uint8_t * dest = pix;
+    uint32_t * src = (uint32_t *)pix;
+    for(size_t y = 0; y < height; y++) {
+        /* Convert a 4 bytes per pixel in format ARGB to R5G6B5A8 format
+            naive way:
+                        r = ((c & 0xFF0000) >> 19)
+                        g = ((c & 0xFF00) >> 10)
+                        b = ((c & 0xFF) >> 3)
+                        rgb565 = (r << 11) | (g << 5) | b
+                        a = c >> 24;
+            That's 3 mask, 6 bitshift and 2 or operations
+
+            A bit better:
+                        r = ((c & 0xF80000) >> 8)
+                        g = ((c & 0xFC00) >> 5)
+                        b = ((c & 0xFF) >> 3)
+                        rgb565 = r | g | b
+                        a = c >> 24;
+            That's 3 mask, 3 bitshifts and 2 or operations */
+        for(size_t x = 0; x < width; x++) {
+            uint32_t in = src[x];
+            uint16_t r = (uint16_t)(((in & 0xF80000) >> 8) | ((in & 0xFC00) >> 5) | ((in & 0xFF) >> 3));
```

Comment:
You can use `lv_color_make(r, g, b)`. It handles `LV_COLOR_SWAP_16` too.

Reply:
Well, it's not the same semantic here. `lv_color_make` creates a triplet of bytes (R, G, B) shifted by the required amount and then it need to be concatenated (even if it's hidden by the bitfield, the compiler will do it). Writing this way is like the naive way commented above, it's not efficient. 
Here, I'm making the mask & shift & concatenation at the same time so it's a lot faster. 

To give an insight, with my code, the cumulative clock count is 6650 clocks, while with this code:
```
lv_color_t c = lv_color_make((in & 0xFF0000) >> 24, (in & 0xFF00) >> 16, (in & 0xFF));
uint16_t r = c.full;
```
it's 13318 clocks, so ~ twice slower.
Since we are working on a large buffer, the impact is far from negligible.

Reply:
Ah, I see. We have `lv_color_hex(in)` too which seems to fit better.

Reply:
`lv_color_hex` is also not good. It expands almost exactly to the code line above and still call `lv_color_make`.
 
While looking at the code, it seems a more efficient `lv_color_hex` could be written (using my trick above to merge shift & mask at the same place and get a x2 performance boost), but I think it'll have to be in its own PR, since it'll imply everything calling it, not only this code.

Reply:
For the sake of simplicity let's stick with your version :slightly_smiling_face: 

The was hell until we could come up with color manipulation macros that make all compiler happy. `lv_color_hex` is not really used in performance-critical places so I wouldn't touch it. 

Just please add a different case for `LV_COLOR_SWAP_16 == 1`. In this case, the endianness of RGB565 is changed.

Reply:
Done.